plugins/utils/auto-session.nix

```diff
@@ -0,0 +1,187 @@
+{
+  pkgs,
+  config,
+  lib,
+  ...
+}:
+with lib; let
+  cfg = config.plugins.auto-session;
+  helpers = import ../helpers.nix {inherit lib;};
+in {
+  options.plugins.auto-session =
+    helpers.extraOptionsOptions
+    // {
+      enable = mkEnableOption "auto-session";
+
+      package = helpers.mkPackageOption "auto-session" pkgs.vimPlugins.auto-session;
+
+      logLevel =
+        helpers.defaultNullOpts.mkEnum
+        ["debug" "info" "warn" "error"]
+        "error"
+        "Sets the log level of the plugin.";
+
+      autoSession = {
+        enable = helpers.defaultNullOpts.mkBool true ''
+          Enables/disables auto creating, saving and restoring.
+        '';
+
+        enableLastSession = helpers.defaultNullOpts.mkBool false ''
+          Whether to enable the "last session" feature.
+        '';
+
+        rootDir =
+          helpers.defaultNullOpts.mkNullable
+          (with types; either str helpers.rawType)
+          "{__raw = \"vim.fn.stdpath 'data' .. '/sessions/'\";}"
+          ''
+            Root directory for session files.
+            Can be either a string or lua code (using `{__raw = 'foo';}`).
+          '';
+
+        createEnabled = helpers.mkNullOrOption types.bool ''
+          Whether to enable auto creating new sessions
+        '';
+
+        suppressDirs = helpers.mkNullOrOption (with types; listOf str) ''
+          Suppress session create/restore if in one of the list of dirs.
+        '';
+
+        allowedDirs = helpers.mkNullOrOption (with types; listOf str) ''
+          Allow session create/restore if in one of the list of dirs.
+        '';
+
+        useGitBranch = helpers.mkNullOrOption types.bool ''
+          Include git branch name in session name to differentiate between sessions for different
+          git branches.
+        '';
+      };
+
+      autoSave = {
+        enable = helpers.defaultNullOpts.mkNullable types.bool "null" ''
+          Whether to enable auto saving session.
+        '';
+      };
+
+      autoRestore = {
+        enable = helpers.defaultNullOpts.mkNullable types.bool "null" ''
```

Comment:
`enable` or `enabled` ?

Reply:
I have changed to `enabled` to stick to the upstream naming.